### ISSUE TITLE
batch input of multiple texts

### ISSUE THREAD

**Issue by WahaD1123** [iSEE-Laboratory/ReferDINO#11]
Your work is excellent! It seems that Referdino currently mainly supports one text corresponding to one video. How should I handle the batch input of multiple texts for Referdino? Thank you for your answer.

**Comment by tmliang** [comment]
Hi, our inference currently only supports `batch_size=1`. If you want to support batched text input, you would need to duplicate the video input by `batch_size` times, which would lead to significant GPU memory usage (since the number of video frames during inference is often very large). Therefore, for multiple texts, we recommend using iteration to process each text individually, like 

```
with torch.no_grad():
    with autocast(args.enable_amp):
        for exp in exps:  # [b,]
            outputs = model.infer(samples, [exp], [target])
            pred_logits.append(outputs["pred_logits"])  # [1, t, q, k]
            pred_masks.append(outputs["pred_masks"])  # [1, t, q, h, w]
            pred_boxes.append(outputs["pred_boxes"])  # [1, t, q, 4]

pred_logits = torch.cat(pred_logits, 0)  # [b, t, q, k]
pred_masks = torch.cat(pred_masks, 0)  # [b, t, q, k]
pred_boxes = torch.cat(pred_boxes, 0)  # [b, t, q, k]
```

I can share the script for visualizing multiple predicted target masks in a single video (e.g., the cover image of this repository) if you need it.

**Comment by WahaD1123** [comment]
Thank you for your reply! I tried to modify the infer implementation for parallel processing, but the effect is far from being as good as that of iteration. If you have code for processing multi-text inputs, could you please send it to me? Thank you very much!

**Comment by tmliang** [comment]
Unfortunately, I haven't implemented parallel processing for multi-text input. 😢

**Comment by WahaD1123** [comment]
okok, it's fine. Is the script for visualizing multiple predicted target masks in a single video also processed serially?

**Comment by tmliang** [comment]
Yes, I think you might have already implemented it.😂

**Comment by WahaD1123** [comment]
😂okok,do you also input texts serially during the training process?

**Comment by tmliang** [comment]
No, the training supports batched inputs, since the sampling frame number in training is controllable (e.g., 6 frames per video). Maybe you can directly call the `forward` function instead of model.infer() to support batched inference? But it might result in OOM if you use too many frames.
